src/main/java/com/patchworkmc/event/EventHandlerRewriter.java

```diff
@@ -12,37 +12,54 @@
 import org.objectweb.asm.Type;
 
 import com.patchworkmc.Patchwork;
-
-public class EventHandlerScanner extends ClassVisitor {
-	private static final String REGISTER_STATIC = "patchwork_registerStaticEventHandlers";
-	private static final String REGISTER_INSTANCE = "patchwork_registerInstanceEventHandlers";
-	private static final String METAFACTORY_OWNER = "java/lang/invoke/LambdaMetafactory";
-	private static final String METAFACTORY_DESCRIPTOR = "(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;Ljava/lang/invoke/MethodType;Ljava/lang/invoke/MethodHandle;Ljava/lang/invoke/MethodType;)Ljava/lang/invoke/CallSite;";
-	private static final String EVENT_BUS = "net/minecraftforge/eventbus/api/IEventBus";
-
-	private Consumer<EventBusSubscriber> subscriberConsumer;
-	private Consumer<SubscribeEvent> subscribeEventConsumer;
-	private List<SubscribeEvent> staticSubscribers;
+import com.patchworkmc.Constants.EventBus;
+import com.patchworkmc.Constants.Lambdas;
+
+public class EventHandlerRewriter extends ClassVisitor {
+	private final Consumer<EventBusSubscriber> subscriberConsumer;
+	private final Consumer<SubscribeEvent> subscribeEventConsumer;
+	// boolean is "isStatic"
+	private final HashMap<SubscribeEvent, Boolean> subscribeEvents = new HashMap<>();
```

Comment:
todo: just put this in the SubscribeEvent class instead of needlessly doing it here